pkg/rule/rule.go

```diff
@@ -33,11 +37,109 @@ func (g Group) OriginalFile() string {
 	return g.originalFile
 }
 
+func (g *Group) GetRules() []rules.Rule {
+	var rs []rules.Rule
+	for _, rule := range g.Rules() {
+		if r, ok := rule.(*rules.AlertingRule); ok {
+			rs = append(rs, &AlertingRule{r, g.PartialResponseStrategy})
+		} else if r, ok := rule.(*rules.RecordingRule); ok {
+			rs = append(rs, &RecordingRule{r, g.PartialResponseStrategy})
+		}
+	}
+	return rs
+}
+
+func (g Group) GetAlertingRules() []*AlertingRule {
+	var alerts []*AlertingRule
+	for _, rule := range g.Rules() {
+		if alertingRule, ok := rule.(*rules.AlertingRule); ok {
+			alerts = append(alerts, &AlertingRule{alertingRule, g.PartialResponseStrategy})
+		}
+	}
+	sort.Slice(alerts, func(i, j int) bool {
+		return alerts[i].State() > alerts[j].State() ||
+			(alerts[i].State() == alerts[j].State() &&
+				alerts[i].Name() < alerts[j].Name())
+	})
+	return alerts
+}
+
 type AlertingRule struct {
 	*rules.AlertingRule
 	PartialResponseStrategy storepb.PartialResponseStrategy
 }
 
+func (r *AlertingRule) HTMLSnippet(pathPrefix string) html_template.HTML {
```

Comment:
Shouldn't we document this and other new public functions?

Reply:
I am not sure, these functions are only used in UI, I will add some comments.